### PR TITLE
Fix basic-auth on re-run to disable basic auth

### LINKED ISSUE #8: basic auth does not get disabled after enabling it first for openfaas app
## Expected Behaviour

We want to disable basic-auth on openfaas when installing it using k3sup. So any request to URL **/system/functions** can be sent without basic **Authorization** header.
## Current Behaviour

k3sup install is not disabling basic-auth for openfaas app, even after passing the argument to do disable it. It throws error invalid credentials.
## Steps to Reproduce (for bugs)

1. Download k3sup bash script `sudo curl -sLS https://get.k3sup.dev | sh`
2. Install k3sup `sudo install k3sup /usr/local/bin/`
3. Run k3sup `k3sup install --local --ip 127.0.01 --user ubuntu`
4. Install openfaas application `k3sup app install openfaas` will install openfaas with basic auth enabled
5. Again install openfaas application `k3sup app install openfaas --basic-auth=false` which should disable basic auth
5. send curl request shown below, which gives error **_invalid credentials_**
```
curl -X POST \
  http://127.0.0.1:31112/system/functions \
  -d '{
    "service": "test-sleep",
    "image": "functions/alpine:latest",
    "network": "func_functions",
    "envProcess": "sleep 5"
}'
```

## Your Environment

* What Kubernetes distribution are you using (for `k3sup app`)?

```
kubectl version
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}

Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2+k3s1", GitCommit:"cdab19b09a84389ffbf57bebd33871c60b1d6b28", GitTreeState:"clean", BuildDate:"2020-01-27T18:09:26Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}

* What OS or type or VM are you using for your cluster? Where is it hosted? (for `k3sup install/join`):

* Operating System and version (e.g. Linux, Windows, MacOS):

```
uname -a

cat /etc/os-release
```
Linux ubuntu-02-feb 4.15.0-76-generic alexellis/k3sup#86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

### PR DESCRIPTION
**Pull request by Waterdrips**
## Description

Enable switching basic auth on/off on re-running arkade install

Fixes #8

This has been tested by running this build with basic auth on, then off.
Then on, then off. IT was enabled/disabled each time and port forwarding
to the gateway gave the correct auth enabled/disabled status depending on
how it was deployed.

Signed-off-by: Alistair Hey <alistair@heyal.co.uk>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
#8

## How Has This Been Tested?

See description

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: cmd/apps/openfaas_app.go [hunk 1]
```diff
@@ -96,38 +98,35 @@ func MakeInstallOpenFaaS() *cobra.Command {
...
-		pass, _ := command.Flags().GetString("basic-auth-password")
+		if basicAuthEnabled {
+			pass, _ := command.Flags().GetString("basic-auth-password")
 
-		if len(pass) == 0 {
-			var err error
-			pass, err = password.Generate(25, 10, 0, false, true)
-			if err != nil {
-				return err
+			if len(pass) == 0 {
```
Comment (alexellis) [inline cmd/apps/openfaas_app.go:111]: Might be worth checking for `Changed` on the flag here.

File: cmd/apps/openfaas_app.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ func MakeInstallOpenFaaS() *cobra.Command {
 			return fmt.Errorf(`to override the "openfaas", install OpenFaaS via helm manually`)
 		}
 
+		basicAuthEnabled, _ := command.Flags().GetBool("basic-auth")
```
Comment (alexellis) [inline cmd/apps/openfaas_app.go:73]: For bools we need to start handling the error with 

```
if err != nil {
   return err
}
```

File: cmd/apps/openfaas_app.go [hunk 3]
```diff
@@ -154,32 +153,18 @@ func MakeInstallOpenFaaS() *cobra.Command {
...
-
 		clusterRole, _ := command.Flags().GetBool("clusterrole")
-
-		clusterRoleVal := "false"
-		if clusterRole {
-			clusterRoleVal = "true"
-		}
-
 		directFunctions, _ := command.Flags().GetBool("direct-functions")
-		directFunctionsVal := "true"
```
Comment (alexellis) [inline cmd/apps/openfaas_app.go:170]: Are you sure that the behaviour replaced here is the same as before?

Comment (Waterdrips) [reply cmd/apps/openfaas_app.go:170]: fairly sure - the command printed to console

```sh 
Command: /home/heyal/.arkade/bin/helm [upgrade --install openfaas openfaas/openfaas --namespace openfaas --wait --values /tmp/charts/openfaas/values.yaml --set openfaasImagePullPolicy=IfNotPresent --set faasnetes.imagePullPolicy=Always --set basicAuthPlugin.replicas=1 --set ingressOperator.create=false --set queueWorker.maxInflight=1 --set clusterRole=false --set gateway.directFunctions=true --set operator.create=false --set gateway.replicas=1 --set queueWorker.replicas=1 --set basic_auth=true --set serviceType=NodePort]

```

and the raw map printed 
```sh 
map[basicAuthPlugin.replicas:1 basic_auth:true clusterRole:false faasnetes.imagePullPolicy:Always gateway.directFunctions:true gateway.replicas:1 ingressOperator.create:false openfaasImagePullPolicy:IfNotPresent operator.create:false queueWorker.maxInflight:1 queueWorker.replicas:1 serviceType:NodePort]

```

**Comment by Waterdrips** [comment]
requested changes implemented

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**